kestra-io/kestra
# fix(ui): Resolves filename truncation if file contains more than one dot in Namespace Files Editor

Closes #5956 

### Changes

* A logical error was in `ui/src/components/inputs/EditorSidebar.vue` while parsing `filename` & `extension` from the file path.
* Logic to parse `filename` & `extension` from file path is changed in stored in a separate function, that's then accessed wherever we have to extract the file details.
* For removing the redundancy, a similar sort of local function `separateString` is replaced by the new function that contains the logic for parsing `filename` & `extension`.
* For getting extension part of the file, `.split(".")[1]` is replaced with `.split(".").pop()` at various places.

### QA

* Uploaded a file with multiple dots `test.0.0.0.0.0.0.0.txt`, when the file is uploaded, it was visible with same name and when the page is reloaded, it was still with the same name.
* Created files with multiple dots within the folder and and outside the folder, file names are visible without any changes
![image](https://github.com/user-attachments/assets/808e8fef-d0e3-4d00-a385-c763cba0d3bb)
* Ran simple flow to verify file read operations 
![image](https://github.com/user-attachments/assets/7adfb6cd-033e-4c35-8964-f7c39283d68c)

* Flow 
```yaml
id: pip_namespace_files
namespace: company.sales
inputs:
  - id: date
    type: STRING
    defaults: "12/24/2024"
    displayName: Delivery Date
tasks:
  - id: python
    type: io.kestra.plugin.scripts.python.Script
    namespaceFiles:
      enabled: true
    script: |
      with open("test/test.0.0.0.0.0.0.0.txt") as file:
        print(file.read())
      
      with open("test/test_inner/test.ef.txt") as file:
        print(file.read())
      
      with open("out.0.0.1.side.txt") as file:
        print(file.read())

```
